mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -405,12 +405,14 @@ private[ml] object DefaultParamsWriter {
   def saveMetadata(
       instance: Params,
       path: String,
-      sc: SparkContext,
```

Comment:
I wonder if we should retain the existing SparkContext method. Third party libraries would still use it, unless they later make a change like you're making to Spark ML. So this would break them. When it's easy enough to retain (and deprecate?) this shared method. Same for loadMetadata

Reply:
Yes this might be an issue, e.g. xgboost4j-spark lib is influenced:
https://github.com/dmlc/xgboost/blob/d6385355815005625115b648f6b7dc861eacd47e/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala#L494

Reply:
But xgboost4j-spark is not used widely, we are going to deprecate and remove it from databricks runtime.

Reply:
I don't think that much is directly relevant to Spark; there is just no reason to break compatibility when it's easy to keep the existing method too

Reply:
@srowen @WeichenXu123 I would appreciate it if, as we move towards Spark 4.0, we could all consider the compatibility of third-party libraries beyond those that Databricks plans to deprecate

Reply:
@sergioschena-db ping for updates on this PR

Reply:
@srowen i didn't have the chance to progress on this one.